packages/gatsby/cache-dir/prefetcher.js

```diff
@@ -1,23 +1,35 @@
-module.exports = ({ getNextQueuedResources, createResourceDownload }) => {
+// @flow
+
+type PrefetcherOptions = {
+  fetchNextResource: () => ?Promise,
```

Comment:
since the imports return promises now we can remove the need to call back when a Resource updates. This keeps tracking of the current resource and clears it when the promise resolves/rejects and then tries the next one.